internal/zinc-persist/src/main/scala/sbt/internal/inc/RelationsTextFormat.scala

```diff
@@ -0,0 +1,128 @@
+package sbt.internal.inc
+
+import java.io.{ File, BufferedReader, Writer }
+
+import sbt.internal.util.Relation
+import xsbti.api.DependencyContext._
+
+trait RelationsTextFormat extends FormatCommons {
+
+  def mappers: AnalysisMappers
+
+  private case class Descriptor[A, B](
+    header: String,
+    selectCorresponding: Relations => Relation[A, B],
+    keyMapper: Mapper[A],
+    valueMapper: Mapper[B]
+  )
+
+  private def stringsDescriptor(header: String, rels: Relations => Relation[String, String]) =
+    Descriptor(header, rels, Mapper.forString, Mapper.forString)
+
+  private val allRelations: List[Descriptor[_, _]] = {
```

Comment:
I agree, and also it's better to keep PRs short, otherwise it's very difficult to reason about the changes and we should be a little bit conservative and test everything carefully before a merge :smile:.
  
I'll take care of cleaning up the relations logic and maybe then we can document this code.